src/classes/Provider.php

```diff
@@ -151,82 +151,109 @@ public static function Video($file){
         readfile($path);
     }
 
-	public static function thumb($file){
-        require_once dirname(__FILE__).'/../phpthumb/ThumbLib.inc.php';
+    public static function thumb($file)
+    {
+        require_once dirname(__FILE__).'/../phpthumb/phpthumb.class.php';
 
         $path = File::r2a(File::a2r($file),Settings::$thumbs_dir);
 
         // We check that the thumb already exists, was created after the image, at the right size
         $goodThumb = false;
-        if(file_exists($path) && filectime($file) < filectime($path) ){
-        	$dim = getimagesize($path);
-        	$goodThumb = ($dim[0] == Settings::$thumbs_size && $dim[1] == Settings::$thumbs_size );
+        if (file_exists($path) && filectime($file) < filectime($path)) {
+            $dim = getimagesize($path);
+            $goodThumb = ($dim[0] == Settings::$thumbs_size && $dim[1] == Settings::$thumbs_size );
         }
 
-        if( !$goodThumb ){
-        	/// If we need to create a thumb, then this is a new picture
+        /// If we need to create a thumb, then this is a new picture
+        if (!$goodThumb) {
 
-        	if(Judge::is_public($file)){
-	        	$r = new RSS(Settings::$conf_dir."/photos_feed.txt");
-	        	$webpath = Settings::$site_address."?f=".urlencode(File::a2r($file));
-	        	$r->add(basename($file),$webpath, "<img src='$webpath&t=Thb' />");
-	        }
+            if (Judge::is_public($file)) {
+                $r = new RSS(Settings::$conf_dir."/photos_feed.txt");
+                $webpath = Settings::$site_address."?f=".urlencode(File::a2r($file));
+                $r->add(basename($file),$webpath, "<img src='$webpath&t=Thb' />");
+            }
 
             /// Create directories
-            if(!file_exists(dirname($path))){
+            if (!file_exists(dirname($path))) {
                 @mkdir(dirname($path),0750,true);
             }
 
             /// Create thumbnail
-			$thumb = PhpThumbFactory::create($file);
-			$thumb->adaptiveResize(Settings::$thumbs_size, Settings::$thumbs_size);
+            $thumb = new phpthumb();
+            if (!empty(Settings::$imagemagick_path)) {
+                $thumb->config_imagemagick_path = Settings::$imagemagick_path;
+            }

```

Comment:
if we do provide default settings through Setting.php setting, this will triggered and phpthumb will go into imagick mode… not sure how it behaves, if no imagick installation exists. will test this…